app/controllers/api/files_controller.rb

```diff
@@ -16,7 +16,7 @@ def show
 
       unless content
         begin
-          content = Git::Exercise.for_solution(solution).read_file_blob(params[:filepath])
+          content = solution.read_file(params[:filepath])
```

Comment:
I've opted for introducing a `read_file` method on the `Solution` class, which itself can then handle the distinction between real files and our dynamically created files (like `README.md`). Let me know if this makes sense.

Reply:
Yeah - I like this.